src/main/java/com/google/sps/servlets/userServlet.java

```diff
@@ -10,13 +19,62 @@
 @WebServlet("/userinformation")
 public class userServlet extends HttpServlet {
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    //dummy data for user
+    String username = "Sam";
+    
     Gson gson = new Gson();
-    MockUserData mockUserData = new MockUserData();
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("User");
+    
+    String userJson = "";
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    for(Entity entity : results.asIterable()){
+      if(entity.getProperty("name") == username){
+        String email = (String) entity.getProperty("email");
+        String[] favouritesArr = gson.fromJson((String) entity.getProperty("favourites"), String[].class);
+        List<String> favourites = Arrays.asList(favouritesArr);
+        String[] pastReviewsArr = gson.fromJson((String) entity.getProperty("pastReviews"), String[].class);
+        List<String> pastReviews = Arrays.asList(pastReviewsArr);
 
-        response.setContentType("text/html;");
-        response.getWriter().println("<h1>Hello world!</h1>");
+        User user = new User(username, email, favourites, pastReviews);
+        userJson = String.format("{\"data\" : %s, \"id\": %s }", gson.toJson(user), entity.getKey().getId());
       }
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(userJson);
+  }
+
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    List<String> favourites = Arrays.asList("1","2","3","4");
+    List<String> pastReviews = Arrays.asList("5","6","7","8");
```

Comment:
If we're storing reviews under the user entities, I think it's better to store a list of keys to the review entities to avoid any duplication.

Reply:
If this is just meant to be hardcoded for now, it's fine but good to note if you're planning to work on it further!

Reply:
For now, i hardcoded it but i will replace it with the keys to review entities eventually! :)